pulsar-broker/src/test/java/org/apache/pulsar/client/api/SimpleProducerConsumerTest.java

```diff
@@ -606,6 +606,61 @@ public void testSendTimeout(int batchMessageDelayMs) throws Exception {
         log.info("-- Exiting {} test --", methodName);
     }
 
+    @Test(dataProvider = "batch")
+    public void testSendTimeoutAndRecover(int batchMessageDelayMs) throws Exception {
+        log.info("-- Starting {} test --", methodName);
+
+        Consumer<byte[]> consumer = pulsarClient.newConsumer().topic("persistent://my-property/my-ns/my-topic5")
```

Comment:
Since the intention is to test partitioned topics, it would be necessary to create a partitioned topic.
Please check https://github.com/apache/pulsar/blob/22a0beb5d4dc4c57fede791194573535bc7b8b10/pulsar-broker/src/test/java/org/apache/pulsar/client/api/SimpleProducerConsumerTest.java#L2432-L2434 for an example of creating a partitioned topic.

Reply:
Please ensure that the test fails without your fix and applying the fix makes the test pass.

Reply:
Thanks for you review. I have addressed done. PTAL.